src/darwin/Framework/CHIP/CHIPDeviceController.h

```diff
@@ -18,11 +18,11 @@
 #ifndef CHIP_DEVICE_CONTROLLER_H
 #define CHIP_DEVICE_CONTROLLER_H
 
-#import "CHIPDevice.h"
-#import "CHIPDeviceStatusDelegate.h"
-#import "CHIPError.h"
+#import <CHIP/CHIPDeviceStatusDelegate.h>
```

Comment:
You are returning `CHIPDevice*`, which would be from transitive includes. Unsure if you want this within the darwin code or not. I believe clang may have some clang-tidy for these things, but running them may not be as easy.

Reply:
I forward declared this class to avoid the `#include`. 
Normally this might result in the user of this class to be forced to include `CHIPDevice.h` on their own but in Objc the norm is to always use an umbrella header from your dependency. 
So in this case the user of this API is expected to have included `CHIP/CHIP.h` and so wouldn’t need to worry about explicitly importing `CHIPDevice.h` as well.